### PR TITLE
Counter, SearchForm and GenreList

### PR DESCRIPTION
**Pull request by bcvitanusic**
Added counter, searchform and genrelist

### CODE REVIEW

File: glomentorprogram/src/components/Counter/Counter.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+function Button(props) {
+	return el(
+		'button',
+		{ onClick: props.handleClick, className: 'counter-button' },
+		props.name
+	);
+}
+
+class Counter extends React.Component {
```
Comment (leandrocunha) [inline glomentorprogram/src/components/Counter/Counter.js:14]: @bcvitanusic once you are using state you should user a `constructor` in your class component. You can do somenthing like

```
class Counter extends React.Component {
   constructor(props) {
     super(props)
     this.state = { ... }
     this.handleIncrement = this.handleIncrement.bind(this)
     ...
   }
   ...
 }
```

You can check more [details here](https://legacy.reactjs.org/docs/react-component.html#constructor)

File: glomentorprogram/src/components/SearchForm/SearchForm.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
+import './SearchForm.css';
+
+function SearchForm({ initialSearchQuery, onSearch }) {
+	const [searchQuery, setSearchQuery] = useState(initialSearchQuery);
```
Comment (leandrocunha) [inline glomentorprogram/src/components/SearchForm/SearchForm.js:5]: @bcvitanusic although this is working one of the requirements is to pass the initial value from props. You are using the `initialSearchQuery` in the placeholder property instead to use into `defaultValue` property.

Take a [look here](https://react.dev/reference/react-dom/components/input#providing-an-initial-value-for-an-input) to check in details

File: glomentorprogram/src/components/GenreSelect/GenreSelect.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React, { useState } from 'react';
+import './GenreSelect.css';
+
+const GenreSelect = ({ GenreList }) => {
```
Comment (leandrocunha) [inline glomentorprogram/src/components/GenreSelect/GenreSelect.js]: @bcvitanusic the requirement for this implementation is to pass 3 props: genreslist, current selected and callback and you pass only one. Please review the requirements to fix it

**Review by leandrocunha — CHANGES_REQUESTED** [review]

File: glomentorprogram/src/components/Counter/Counter.js (hunk 1 above)
Comment (bcvitanusic) [reply glomentorprogram/src/components/Counter/Counter.js:14]: fixed

File: glomentorprogram/src/components/SearchForm/SearchForm.js (hunk 2 above)
Comment (bcvitanusic) [reply glomentorprogram/src/components/SearchForm/SearchForm.js:5]: fixed

File: glomentorprogram/src/components/GenreSelect/GenreSelect.js (hunk 3 above)
Comment (bcvitanusic) [reply glomentorprogram/src/components/GenreSelect/GenreSelect.js]: fixed, passed 3 props

File: glomentorprogram/src/components/Counter/Counter.js (hunk 1 above)
Comment (leandrocunha) [reply glomentorprogram/src/components/Counter/Counter.js:14]: @bcvitanusic Awesome, just a quick comment that there is a warning regarding missing keys in your Counter component. The keys are the important props that React uses as part of your algorithm to check what should be changed.

So you should pass the key props in the children array to fix it. Something like:

```
render() {
  return (
     el('div', { key: 0, className: 'container' }, [ ... ]
  )
}
```

More details about keys you can [find here](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key)

**Review by leandrocunha — APPROVED** [review]

**Pull request merged**